include/hmap/dynamic-hmap.hpp

```diff
@@ -337,6 +338,12 @@ class DynamicHMap {
 		}
 	}
 
+	[[noreturn]] void keyNotFound(const detail::KeyBase& k) const {
+		std::stringstream msg;
+		msg << "DynamicHMap: key '" << k.key.c_str() << "' not present" << std::endl;
```

Comment:
Pretty sure `.c_str()` is unnecessary here since `k.key` is a `std::string`.
```suggestion
		msg << "DynamicHMap: key '" << k.key << "' not present" << std::endl;
```

This message could be further improved by adding a `virtual std::type_info typeInfo() const` to `KeyTagBase`. and accessing it's `.name()`.